src/coordinated_workers/interfaces/cluster.py

```diff
@@ -359,6 +361,13 @@ def gather_topology(self) -> List[Dict[str, str]]:
 
         return data
 
+    def gather_models(self) -> Set[str]:
+        """Gather the Juju models through which this coordinated-worker solution is distributed."""
+        models: Set[str] = {self._charm.model.name}  # type: ignore
+        for worker_unit in self.gather_topology():
+            models.add(worker_unit["model"])
+        return models
+
```

Comment:
Good catch that this could be distributed across multiple models.  I'm not sure they've actually designed for this anyway, but doesn't hurt to cover it here